flink-connectors/flink-connector-elasticsearch-base/src/main/java/org/apache/flink/streaming/connectors/elasticsearch/table/ElasticsearchConfiguration.java

```diff
@@ -81,16 +80,22 @@ public String getDocumentType() {
 		return config.get(ElasticsearchOptions.DOCUMENT_TYPE_OPTION);
 	}
 
-	public Optional<Integer> getBulkFlushMaxActions() {
-		return config.getOptional(ElasticsearchOptions.BULK_FLUSH_MAX_ACTIONS_OPTION);
+	public int getBulkFlushMaxActions() {
+		int maxActions = config.get(ElasticsearchOptions.BULK_FLUSH_MAX_ACTIONS_OPTION);
+		// convert 0 to -1, because Elasticsearch client use -1 to disable this configuration.
+		return maxActions == 0 ? -1 : maxActions;
 	}
 
-	public Optional<Integer> getBulkFlushMaxSize() {
-		return config.getOptional(ElasticsearchOptions.BULK_FLASH_MAX_SIZE_OPTION).map(MemorySize::getMebiBytes);
+	public long getBulkFlushMaxByteSize() {
+		long maxSize = config.get(ElasticsearchOptions.BULK_FLASH_MAX_SIZE_OPTION).getBytes();
+		// convert 0 to -1, because Elasticsearch client use -1 to disable this configuration.
+		return maxSize == 0 ? -1 : maxSize;
 	}
 
-	public Optional<Long> getBulkFlushInterval() {
-		return config.getOptional(BULK_FLUSH_INTERVAL_OPTION).map(Duration::toMillis);
+	public long getBulkFlushInterval() {
+		long interval = config.get(BULK_FLUSH_INTERVAL_OPTION).toMillis();
+		// convert 0 to -1, because Elasticsearch client use -1 to disable this configuration.
+		return interval == 0 ? -1 : interval;
```

Comment:
Why not return empty in case it is set to 0? I prefer explicitness of `Optional` instead of magic values. Plus we would not need to change the Elasticsearch connector code for that.

Reply:
Because Elasticsearch client has a default value for them (1000 rows, 5mb), we have to use `-1` to disable them, otherwise the client will use the default value.

Reply:
got it.

Reply:
The bridge logic seems weird, why not just let user set `-1` which is synced with ES way.

Reply:
Unfortunately, Duration and MemorySize ConfigOptions doesn't allow negative values, e.g. `-1`. That's why I picked zero. I think zero is used as a disabled value in the many places, e.g. Kafka [batch.size](https://kafka.apache.org/documentation/#batch.size), and [watermark interval](https://ci.apache.org/projects/flink/flink-docs-master/ops/config.html#pipeline-auto-watermark-interval).